x-pack/plugins/ml/public/alerting/jobs_health_rule/register_jobs_health_alerting_rule.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { lazy } from 'react';
+import { i18n } from '@kbn/i18n';
+import { TriggersAndActionsUIPublicPluginSetup } from '../../../../triggers_actions_ui/public';
+import { PluginSetupContract as AlertingSetup } from '../../../../alerting/public';
+import { ML_ALERT_TYPES } from '../../../common/constants/alerts';
+import { MlAnomalyDetectionJobsHealthRuleParams } from '../../../common/types/alerts';
+
+export function registerJobsHealthAlertingRule(
+  triggersActionsUi: TriggersAndActionsUIPublicPluginSetup,
+  alerting?: AlertingSetup
+) {
+  triggersActionsUi.alertTypeRegistry.register({
+    id: ML_ALERT_TYPES.AD_JOBS_HEALTH,
+    description: i18n.translate('xpack.ml.alertTypes.jobsHealthAlertingRule.description', {
+      defaultMessage: 'Alert when anomaly detection jobs experiencing realtime issues.',
+    }),
+    iconClass: 'bell',
+    documentationUrl(docLinks) {
+      return `${docLinks.ELASTIC_WEBSITE_URL}guide/en/machine-learning/${docLinks.DOC_LINK_VERSION}/ml-configuring-alerts.html`;
+    },
+    alertParamsExpression: lazy(() => import('./anomaly_detection_jobs_health_rule_trigger')),
+    validate: (alertParams: MlAnomalyDetectionJobsHealthRuleParams) => {
+      const validationResult = {
```

Comment:
If I don't have any jobs or groups, I am still able to select `*` for include jobs but my rule will immediately error with `Rule does not have associated jobs.`. Should there be a validation check to prevent creating this rule if there are no jobs?

Or if the intent is that you can create this health check and it will catch future jobs/groups, maybe it shouldn't error out and just log a warning if there are no jobs?

Reply:
I changed to logging a warning instead of throwing an error in https://github.com/elastic/kibana/commit/8c7b6e669eb2955098fd0d94eb6eac3a6918235d